netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -84,6 +88,9 @@
   public static final AttributeKey<SSLSession> SSL_SESSION_ATTR_KEY =
       AttributeKey.valueOf(SSLSession.class, "ssl-session");
 
+  @VisibleForTesting
+  static boolean validateHeaders = false;
```

Comment:
Actually it doesn't  Hardcoding true or false into the call below had no appreciable performance difference.

Any difference is not possible to measure.